BlueprintUI/Sources/Layout/Alignment.swift

```diff
@@ -0,0 +1,106 @@
+import CoreGraphics
+
+/// Types used to identify alignment guides.
+///
+/// Types conforming to `AlignmentID` have a corresponding alignment guide value,
+/// typically declared as a static constant property of `HorizontalAlignment` or 
+/// `VerticalAlignment`.
+///
+public protocol AlignmentID {
+
+    /// Returns the value of the corresponding guide, in `context`, when not
+    /// otherwise set in `context`.
+    static func defaultValue(in context: ElementDimensions) -> CGFloat
+}
+
+/// An alignment position along the horizontal axis.
+public struct HorizontalAlignment: Equatable {
+
+    var id: AlignmentID.Type
+
+    /// Creates an instance with the given ID.
+    ///
+    /// Note: each instance should have a unique ID.
+    public init(_ id: AlignmentID.Type) {
+        self.id = id
+    }
+
+    public static func == (lhs: HorizontalAlignment, rhs: HorizontalAlignment) -> Bool {
```

Comment:
Could this be synthesized automatically if the `id` parameter was an `ObjectIdentifier`, which was automatically synthesized to return `ObjectIdentifier(type(of: self))`?

Reply:
Er, no coffee yet, I misspoke on this one

If the ID was an `ObjectIdentifier` that was set in init from `AlignmentID.Type`.

Not a big deal since it's only one property, though.

Reply:
I think the only reason that won't work is that we need the metatype to bridge back & forth to `StackLayout.Alignment`. I experimented with making stacks generic around the alignment type but it turned into a much bigger change.